src/Entity/AbstractVehicle.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace App\Entity;
+
```

Comment:
you have brand model and year there, methods for getting them there and also same things in car controller. Move all common parts in abstractVehicle (everything except orm config before class car extends ... )